File: app/src/main/java/com/futsch1/medtimer/RepeatRemindersPreferencesFragment.java
package com.futsch1.medtimer;

import android.os.Bundle;

import androidx.preference.PreferenceFragmentCompat;

public class RepeatRemindersPreferencesFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.repeat_reminders_preferences, rootKey);
    }
}

File: feature/ui/src/main/java/com/futsch1/medtimer/feature/ui/preferences/RepeatRemindersPreferencesFragment.kt
package com.futsch1.medtimer.feature.ui.preferences

import android.os.Bundle
import com.futsch1.medtimer.core.ui.preferences.MedTimerPreferenceFragment
import com.futsch1.medtimer.core.datastore.PreferencesDataSource
import com.futsch1.medtimer.feature.ui.R
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class RepeatRemindersPreferencesFragment : MedTimerPreferenceFragment() {
    @Inject
    lateinit var preferencesDataSource: PreferencesDataSource

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        preferenceManager.preferenceDataStore = preferencesDataSource

        setPreferencesFromResource(R.xml.repeat_reminders_preferences, rootKey)
    }
}
